File: 3. Linked List/10_remove_node/10_remove_node.java
class Node<T> {
    T val;
    Node<T> next;

    public Node(T val) {
        this.val = val;
        this.next = null;
    }
}

class Source {
    public static <T> Node<T> removeNode(Node<T> head, T targetVal) {
        if (head.val == targetVal) {
            return head.next;
        }

        Node<T> prev = null;
        Node<T> current = head;

        while (current != null) {
            if (current.val == targetVal) {
                prev.next = current.next;
                break;
            }
            prev = current;
            current = current.next;
        }

        return head;
    }
}

File: 3. Linked List/10_remove_node/10_remove_node.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def remove_node(head, target_val):
    if head.val == target_val:
        return head.next

    prev = None
    current = head

    while current is not None:
        if current.val == target_val:
            prev.next = current.next
            break
        prev = current
        current = current.next

    return head
